bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -276,20 +274,93 @@ private void manageCurPosIfEnvContainsTarget(final SPTarget target, final Runnab
             t.addWatcher(posWatcher);
             refreshAll();
         });
+        return true;
+    }
+
+    /**
+     * Determine whether or not the auto group has changed for a specific observation.
+     */
+    private boolean autoGroupChanged(final ISPObsComponent oldObsComp, final TargetObsComp oldTOC) {
+        final TargetObsComp newTOC = getDataObject();
+
+        // If either of the TargetObsComps are null, this was not triggered by an auto group change.
+        if (oldTOC == null || newTOC == null) {
+            return false;
+        }
+
+        // If either of the node keys are null, or they are defined but different, this is not an auto group change.
+        final SPNodeKey oldNodeKey = ImOption.apply(oldObsComp).map(ISPObsComponent::getNodeKey).getOrNull();
+        final SPNodeKey newNodeKey = ImOption.apply(getNode()).map(ISPObsComponent::getNodeKey).getOrNull();
+        if (oldNodeKey == null || newNodeKey == null) {
+            return false;
+        }
+        if (!oldNodeKey.equals(newNodeKey)) {
+            return false;
+        }
+
+        // Extract the target environments and check if the auto groups have changed.
+        final TargetEnvironment oldEnv    = oldTOC.getTargetEnvironment();
+        final TargetEnvironment newEnv    = newTOC.getTargetEnvironment();
+        final Option<GuideGroup> oldGpOpt = oldEnv.getGroups().headOption().filter(GuideGroup::isAutomatic);
+        final Option<GuideGroup> newGpOpt = newEnv.getGroups().headOption().filter(GuideGroup::isAutomatic);
```

Comment:
We should probably add a `getAutomaticGroup` to `GuideEnvironment` so that you don't have to do this.  `GuideEnv` keeps a reference to the automatic group that can just be returned.  Something like:

````scala
def getAutomaticGroup: GuideGroup = {
   GuideGroup(guideEnv.auto)
}
````